php-checks/src/main/java/org/sonar/php/checks/utils/CheckUtils.java

```diff
@@ -107,6 +107,39 @@ public static boolean areSyntacticallyEquivalent(AstNode node1, AstNode node2) {
     }
     return true;
   }
+  
+  public static boolean areSyntacticallyEquivalent(@Nullable Tree tree1, @Nullable Tree tree2) {
+    if (tree1 == tree2) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)